tomato/views.py

```diff
@@ -292,27 +294,136 @@ def statistics(self, request):
         start = parse_datetime(request.query_params.get('start_date', None))
         end = parse_datetime(request.query_params.get('end_date', None))
 
-        if start and end and end > start:
-            all_queryset = Tomato.objects.filter(
-                acquisition_date__lte=end,
-                acquisition_date__gte=start,
-            )
-            user_queryset = all_queryset.filter(user=request.user)
-            totals = {
-                "global": self.get_queryset_number_of_tomatoes(all_queryset),
-                "user": self.get_queryset_number_of_tomatoes(user_queryset)
-            }
-            return Response(
-                {
-                    "totals": totals,
-                },
-                status=status.HTTP_200_OK,
-            )
-
-        else:
+        if not start or not end or end < start:
             return Response(
                 {
                     'dates': _('Please select a valid start and end dates'),
                 },
                 status=status.HTTP_400_BAD_REQUEST,
             )
+
+        return Response(
+            {
+                "totals": self._get_total_data(
+                    start_date=start,
+                    end_date=end,
+                ),
+                "graph": self._get_graph_data(
+                    start_date=start,
+                    end_date=end,
+                )
```

Comment:
I see the ChartJS functions from SDT. Should we also inherit this view from a Abstract view, maybe created in utils.py so If we need other graph later or need to extract this code in the future it's more easy ? And for the fields used in the _functions we can set them as view attributes instead ?
Speaking about _get_graph_data, _define_interval, _trunc_interval, _get_intervals, _get_data